x-pack/plugins/data/public/plugin.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreSetup, CoreStart, Plugin } from 'src/core/public';
+import { DataPublicPluginStart, DataPublicPluginSetup } from '../../../../src/plugins/data/public';
+import { setAutocompleteService } from './services';
+import { setupKqlQuerySuggestionProvider, KUERY_LANGUAGE_NAME } from './autocomplete';
+
+export interface DataPublicSetupDependencies {
+  data: DataPublicPluginSetup;
+}
+export interface DataPublicStartDependencies {
+  data: DataPublicPluginStart;
+}
+
+export type DataPublicSetup = ReturnType<DataPublicPlugin['setup']>;
```

Comment:
I think it's not a good idea to name the types \ interfaces \ plugins for public and server in both OSS and X-Pack in exactly the same names.

I do think this should be `DataEnhancedSetup` etc. 

For more reference, you can take a look at the naming generated by the [plugin generator](https://github.com/elastic/kibana/pull/55281) (soon to be merged)

Reply:
make sense, prefix `DataEnhanced` was added

Reply:
agreed, let's keep the names consistent. 

the only caveat i would add is that as was also discussed in the same PR Liza links to (#55281), the naming convention `*PublicPlugin` should just be `*Plugin` per the [CONVENTIONS.md](https://github.com/elastic/kibana/blob/master/src/core/CONVENTIONS.md)

So this should be `DataEnhancedSetup`, `DataEnhancedStart` interfaces (the same on both client and server)